core/src/main/scala/cromwell/core/ExecutionStatus.scala

```diff
@@ -3,10 +3,11 @@ package cromwell.core
 object ExecutionStatus extends Enumeration {
   type ExecutionStatus = Value
   val NotStarted, Starting, Running, Failed, Preempted, Done, Aborted = Value
+  val TerminalStatuses = Set(Failed, Done, Aborted, Preempted)
```

Comment:
Your change is minor so ignore this if you wish but IMO this was already at "really should be an ADT" level and now we're tacking stuff on to it here.

Instead of the first implicit class and an internal set, there could be a sealed trait ExecutionStatus as well as a sub sealed trait TerminalStatuses with isTerminal being a function on ExecutionStatus.

The one gotcha is the withName in the other implicit class but IMO the improvement would be worth it